### DISCUSSION TITLE
ESP Chip ID & co. as variable?

### DISCUSSION THREAD

**Discussion by 2000ede** [arendst/Tasmota#20311] [Q&A] [answered]
Hello,
is this information available as a system variable?
Or another way to use them in code?

![grafik](https://github.com/arendst/Tasmota/assets/37264959/e28e1c55-2515-4733-a8bc-04382da3afd5)

I only found this one:

System variables (for debugging)
stack = stack size
heap = free heap size
pheap = PSRAM free heap size (ESP32)
core = current core (0 or 1) (ESP32)
ram = used ram size
slen = script length
freq = cpu frequency
micros = running microseconds
millis = running milliseconds
loglvl = loglevel of script cmds
(https://tasmota.github.io/docs/Scripting-Language/#special-variables)

**Comment by sfromis** [comment, marked as answer]
Short answer is no, but many fields (most of what you ask for) are available from various commands like `Status` and can be gathered from the result. Example:
https://tasmota.github.io/docs/Scripting-Language/#retrieve-network-gateway-ip-address

The ESP chip id is the last 3 bytes of the MAC address available from `Status 5`.

**Reply by 2000ede** [reply]
Thanks @sfromis!
